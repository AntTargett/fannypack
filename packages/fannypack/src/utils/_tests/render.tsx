import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
