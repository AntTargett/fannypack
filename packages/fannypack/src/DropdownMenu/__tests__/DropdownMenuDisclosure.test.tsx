import * as React from 'react';
import { DropdownMenu } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Disclosure {...menu}>Actions</DropdownMenu.Disclosure>}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for CSS props', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu.Disclosure {...menu} backgroundColor="red">
            Actions
          </DropdownMenu.Disclosure>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('DropdownMenu.Disclosure.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu.Disclosure
            {...menu}
            overrides={{ DropdownMenu: { Disclosure: { css: { root: { backgroundColor: 'red' } } } } }}
          >
            Actions
          </DropdownMenu.Disclosure>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('DropdownMenu.Disclosure.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Disclosure {...menu}>Actions</DropdownMenu.Disclosure>}
      </DropdownMenu.State>,
      { theme: { DropdownMenu: { Disclosure: { css: { root: { backgroundColor: 'red' } } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => <DropdownMenu.Disclosure {...menu}>Actions</DropdownMenu.Disclosure>}
      </DropdownMenu.State>,
      { theme: { DropdownMenu: { Disclosure: { defaultProps: { className: 'test', color: 'primary' } } } } }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
