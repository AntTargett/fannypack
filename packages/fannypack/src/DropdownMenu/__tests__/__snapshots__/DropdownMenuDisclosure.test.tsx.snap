// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  color: #3722ed !important;
}

<div
  aria-controls="test"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-DropdownMenuDisclosure"
  orientation="vertical"
  role="button"
  tabindex="0"
>
  Actions
</div>
`;

exports[`overrides DropdownMenu.Disclosure.css.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-controls="test"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DropdownMenuDisclosure"
  orientation="vertical"
  role="button"
  tabindex="0"
>
  Actions
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  aria-controls="test"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DropdownMenuDisclosure"
  orientation="vertical"
  role="button"
  tabindex="0"
>
  Actions
</div>
`;

exports[`props should render correctly for CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  background-color: red !important;
}

<div
  aria-controls="test"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DropdownMenuDisclosure"
  orientation="vertical"
  role="button"
  tabindex="0"
>
  Actions
</div>
`;

exports[`theming DropdownMenu.Disclosure.css.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: red;
}

<div
  aria-controls="test"
  aria-expanded="false"
  aria-haspopup="menu"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DropdownMenuDisclosure"
  orientation="vertical"
  role="button"
  tabindex="0"
>
  Actions
</div>
`;
